### DISCUSSION TITLE
Connection stuck(ping packet loss) on the hour even IKEv2 VPN is alive

### DISCUSSION THREAD

**Discussion by wangbin-ouhin** [strongswan/strongswan#1375] [Q&A] [answered]
Hello,

We use strongSwan IKEv2 VPN for about a month, we found connection(ping loss) stuck about 10 seconds on the hour even VPN is alive.
It will recovery automatically. And this problem doesn't happy every hour.
We use ICMP ping to trace the packets and find out the packets wasn't send to the VPN server when problem occurs.

Here is our configuration and log. 

**System**
 - OS: CentOS Linux release 7.9.2009 (Core)
 - strongSwan version(s): Linux strongSwan U5.7.2

**ipsec.conf**

> config setup
        strictcrlpolicy=yes
        uniqueids=yes
conn gate-tokyo
        auto=add
        keyexchange=ikev2
        forceencaps=yes
        ike=aes256-sha256-modp2048,3des-sha1-modp2048,aes256-sha1-modp2048,aes256-sha1-modp1024,aes128-sha1-modp1024,3des-sha1-modp1024!
        esp=aes128-sha256
        dpdaction=restart
        #mobike=no
        dpddelay=300s
        rekey=yes
        left=%any
        leftid=@[vpn server's domain]
        #leftcert=fullchain.pem
        leftcert=server-cert.pem
        leftsendcert=always
        leftsubnet=0.0.0.0/0
        right=%any
        rightid=%any
        rightauth=eap-radius
        rightdns=192.168.100.21
        rightsourceip=192.168.220.128/25
        rightsendcert=never
        eap_identity=%any
        lifetime=1h
        ikelifetime=24h
        rekeyfuzz=0%

**charon.log / iptables-save / ip route show table all / ip address**
[ip route show table all.txt](https://github.com/strongswan/strongswan/files/9867734/ip.route.show.table.all.txt)
[strongswan statusall.txt](https://github.com/strongswan/strongswan/files/9867735/strongswan.statusall.txt)
[iptables-save.txt](https://github.com/strongswan/strongswan/files/9867736/iptables-save.txt)
[charon.log](https://github.com/strongswan/strongswan/files/9867737/charon.log)
[ip address.txt](https://github.com/strongswan/strongswan/files/9867738/ip.address.txt)
> See the attachment

**ICMP ping lost capture**
Client --> server in the VPN subnet : 
![1666773595189](https://user-images.githubusercontent.com/2121401/197978898-0e85a47f-9f6e-40a3-babb-c45a03a150c5.png)

packets of the server in VPN subnet:
![1666773782328](https://user-images.githubusercontent.com/2121401/197979048-8450c4c0-76ef-4150-bff1-e7a12b3150d5.png)

If you need any additional information , please let us know.

Is there anything wrong in our configuration?
Waiting for your reply.
Best regards,

**Comment by Thermi** [comment, marked as answer]
Do a plaintext ping over the hour to check if it's a connectivity issue eventually. I have seen in some networks that the operators sometimes (for some reason) rebuild the BGP routing tables and intermittently when that happens there's no connectivity until the routing tables converge again.

**Reply by Thermi** [reply]
Also don't do that: rekeyfuzz=0%

**Reply by Thermi** [reply]
And maybe capture ESP/UDP port 4500 packets instead, then you can check if you get packets at all. You can check /proc/net/xfrm_stat to see if XFRM encountered problems. Of course check if any of the counters there incremented when the problem occured.

**Reply by wangbin-ouhin** [reply]
Hi Thermi,

Thank you for the reply.

>Do a plaintext ping over the hour to check if it's a connectivity issue eventually. 

Because all of the traffic were passed through the VPN, we didn't test client's connectivity properly.
But we tested two client's VPN connectivity on the same time, when this problem occurs another client worked fine.
So we thought no problem happened in the network in the VPN side.

  
>Also don't do that: rekeyfuzz=0%

Changed rekeyfuzz to default setting(100%).

>And maybe capture ESP/UDP port 4500 packets instead, then you can check if you get packets at all. 

We are capturing the ESP packet now. When it happens again, I will affix the newest information.

**Reply by wangbin-ouhin** [reply]
Hi Thermi,

This problem happened last hour.
Here is the packet capture detail.

**Packet No:1965 --> The last packet successfully sent to server.**
![1666843178081](https://user-images.githubusercontent.com/2121401/198188492-74a688f2-de51-40a4-bcca-a30c50f77f01.png)

**Pakcet No:1966~2054  --> Packets to VPN server were not received on server, and get no packet from server.**
![1666843334665](https://user-images.githubusercontent.com/2121401/198188932-3d27d2ce-991c-4c70-87c6-9dd4c23ac385.jpg)

**Packet No:2055  --> Connection recovered.**

So it looks like something wrong of the client's network connection, is it right?

**Reply by wangbin-ouhin** [reply]
Ping packet to the VPN server lost at 18:00, so looks like the internet connection of the client cause this problem ,right?
![1666861622498](https://user-images.githubusercontent.com/2121401/198242670-6830ed84-30de-4277-b6e6-26410c02100b.png)

**Comment by Devstellar** [comment]
You are using NAT, I would suggest to ping outside IPSec, to the public address, to really discard network problem in transit.

**Reply by wangbin-ouhin** [reply]
Hi Devstellar,

Thank you for your reply.

> I would suggest to ping outside IPSec,

All of the traffic will through the VPN connection when it is on.
Do you mean turn off the VPN  , ping public address , and check if it will lost ping packet on the hour?

**Reply by wangbin-ouhin** [reply]
Ping packet to the VPN server lost at 18:00, so looks like the internet connection of the client cause this problem ,right?
![1666861622498](https://user-images.githubusercontent.com/2121401/198242670-6830ed84-30de-4277-b6e6-26410c02100b.png)

**Reply by Devstellar** [reply]
> Ping packet to the VPN server lost at 18:00, so looks like the internet connection of the client cause this problem ,right? 

Yes. In this case is a Network unreachable msg, so you must look into who is sending this ICMP unreachable message to troubleshoot problem further.
Usually the ICMP unreachable is coming from gateway (router) of client, and this usually means his internet connection went down.

**Reply by Thermi** [reply]
> Ping packet to the VPN server lost at 18:00, so looks like the internet connection of the client cause this problem ,right?

Yes, I guess so.

**Reply by wangbin-ouhin** [reply]
Hi Devstellar, Thermi

Thank you for the reply and advice.

Best regards,
